src/client-lib/dropdown/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classnames from 'classnames';
+
+class Dropdown extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isOpen: false
+        }
+    }
+
+    close() {
+        this.setState({ isOpen: false });
+    }
+
+    componentDidMount() {
+        document.addEventListener('click', this._handleDocumentClick, false);
+    }
+
+    componentWillUnmount() {
+        document.addEventListener('click', this._handleDocumentClick, false);
+    }
+
+    _handleDocumentClick = (event) => {
+        if (!ReactDOM.findDOMNode(this).contains(event.target)) {
+            this.setState({ isOpen: false });
+        }
+    };
+
+    _handleMouseDown = (event) => {
+        if (event.type == 'mousedown' && event.button !== 0) return;
```

Comment:
That's great but it's in our code now, should probably understand what it's doing and why.